secure_message/resources/messages.py

```diff
@@ -34,11 +34,18 @@ def post(self):
 
         if 'msg_id' in post_data:
             raise BadRequest(description="Message can not include msg_id")
+        if post_data.get('thread_id'):
+            conversation_metadata = Retriever.retrieve_conversation_metadata(post_data.get('thread_id'))
+            if conversation_metadata is None:
+                abort(404)
```

Comment:
I think this shuld be more aboort(404, "you message about why di dnot work")

Reply:
It's a 404, so what went wrong should be obvious.  From what I remember there is some default text that get's added when you return it, but I'll double check that now

Reply:
I've removed this now.  I realised that there's going to be a time where there won't be a conversation record for every thread id.